### ISSUE TITLE
Generation of train.txt, val.txt and test.txt for large custom datasets

### ISSUE THREAD

**Issue by SaadRiazQazi** [BVLC/caffe#4308]
I am wondering if there is a feature in caffe that will allow automatic generation of the subjected files for large custom datasets or for datasets that don't come with this information. I have a custom dataset of 500,000 images and use digits to generate these files and then use them in caffe. If there is no such feature, of which I am highly doubtful, it will be an interesting addition.

**Comment by malreddysid** [comment]
Are you talking about automatic segregation of the dataset into training set, testing set and validation set?
